Client/Frontend/Login Management/LoginListViewController.swift

```diff
@@ -166,6 +172,44 @@ class LoginListViewController: UIViewController {
     }
 }
 
+// MARK: - Favicon Loading
+extension LoginListViewController {
+
+    private func loadFaviconsForLogins(logins: [Login]) {
+        logins.forEach { login in
+            loadFaviconsForLoginFromHistory(login).uponQueue(dispatch_get_main_queue()) { result in
```

Comment:
Ah, I see you exclude non-visible rows from the UI part. So the DB work will be the majority — worth batching up.